libraries/chain/transaction_context.cpp

```diff
@@ -172,6 +172,11 @@ namespace eosio { namespace chain {
          deadline_exception_code = billing_timer_exception_code;
       }
 
+      if( !explicit_billed_cpu_time ) {
+         // if account no longer has enough cpu to exec trx, don't try
+         validate_account_cpu_usage( billed_cpu_time_us, account_cpu_limit );
```

Comment:
This doesn't seem to take leeway or the block cpu limit into account.

Reply:
Good catch. @arhag and I discussed that and came to the conclusion that it should be fine that it doesn't take leeway into account.

Reply:
Won't that effectively make leeway useless?

Reply:
This only comes into play on unapplied transactions. So if not applied before then leeway will be in effect. Leeway will just not be used for unapplied transactions.

Reply:
If we start relaying the billed_cpu_time over the p2p network then we will have to factor in leeway somehow.